doc/requirements/Requirements.md

```diff
@@ -0,0 +1,704 @@
+Parameter Framework.
+<why>This can happen when the underlying hardware subsystem
+crashes/reboots/...</why>
+
+When a syncer reports an out-of-sync condition, the PF **MUST** try to resync
+the hardware values.
+
+# Rule based dynamic abstraction
+
+## Philosophy
+
+The PF offers parameters mapped on hardware. This is a good but weak
+abstraction. There is often a 1/1 relation between a parameter and the hardware
+it maps.
+
+A PF offers a mechanism to abstract the parameters to more high level concept.
+
+The goal is to hide numerous parameters and their dynamic value behind simple
+and human friendly API.
+
+It works by regrouping parameters with similar management and defining
+configurations for each "scenario". These "scenario" are then given a priority
+and a detection predicate. Configuration are applied when their associated
+"scenario" is detected.
+
+"Scenario" are detected through arbitrary criterion provided by the PF host
+(see below).
+
+## Definition
+
+<dl>
+<dt>Configuration</dt>
+<dd>
+Set of values for different parameters. A configuration **MUST NOT** contain 2
+values of the same parameters.
+
+For example, given a PF with 3 integer parameters A,B,C, a configuration can
+contain:
+
+ - 1 value: (A) or (B) or (C); or
+ - 2 values: (A,B) or (A,C) or (B,C); or
+ - 3 values: (A,B,C).
+</dd>
+
+<dt>Rogue Parameter</dt>
+<dd>
+A Parameter that is not contained by any configuration.
+<dd>
+</dl>
+
+## Configuration
+
+A PF **MUST** offer configurations as described in the Definition chapter.
+<nb>rule based parameter engine does not manipulate directly values, it
+applies configuration on the parameters.</nb>
+<why>This is what the reference implementation does.</why>
+
+Each configuration **MUST** be associated with a predicate that condition its
+eligibility. A configuration with a predicate that evaluates to `true` is called
+an "eligible configuration"
+<why>This is what the reference implementation does.</why>
+
+It **SHOULD** be possible to express a predicate to always evaluates to `true`.
+<why>in order to have parameters set to constant values or have a fallback
+configuration in a domain -- see below.</why>
+
+The predicate **SHOULD** be a "selection criterion rule". See next chapter for a
+definition.
+<why>The reference implementation uses a boolean expression based engine.</why>
+
+## Selection criterion
+
+A selection criterion **MUST** have one, and only one, state at a given time.
+
+A selection criterion **MUST** know at construction all possible states.
+<why>To be able to validate: -- rules on start -- state changes</why>
+
+The selection criterion possible states **MUST** be specifiable by directly a
+state set (`Input -> states == identity`)
+<nb>called **exclusive criterion**</nb>
+<nb>An empty set is not allowed as the criterion could not have a state.</nb>
+<why>any criterion can be crated with this one.</why>
+
+The selection criterion possible states **SHOULD** be specifiable by a combination
```

Comment:
Because a criterion set is a set of criterionS. Thus not ONE criterion. It would make no sense to define a criterion stateS with multiple criterionS.